### PR TITLE
Pulls languages from multiple repos

### LINKED ISSUE #3805: Allow for project "languages" to be pulled from multiple repositories
### Overview
As developers, we need to ensure all of the languages are displaying on the [Civic Tech Index Project Page](https://www.hackforla.org/projects/civic-tech-index). Since Civic Tech Index has two separate repositories for their front and back end, we need to figure out a way to display the languages data from both repos.

### Action Items
- [x] Understand how the website is currently grabbing the `languages` data for projects
- [x] Understand how the languages data makes it onto each project page (not coming from the [civic-tech-index.md](https://github.com/hackforla/website/blob/gh-pages/_projects/civic-tech-index.md))
- [x] Figure out which file controls the languages that are displayed for each project. Alter the logic of this file so that the Civic Tech Index pages shows the languages from both of their github repositories.

### Resources/Instructions
- [See Comment that created this issue](https://github.com/hackforla/website/issues/2728#issuecomment-1086902818)
- [Wiki: How languages are added to the `project` variable](https://github.com/hackforla/website/wiki/Guide-to-the-Project-Layout-File#adding-languages)
- [CTI Front End Repo](https://github.com/civictechindex/CTI-website-frontend)
- [CTI Back End Repo](https://github.com/civictechindex/CTI-website-backend)

### PR DESCRIPTION
**Pull request by StephenTheDev1001**
Fixes #3805

### What changes did you make?
  -allow for 2 repo ids to be scraped
  -merge lanaguages array from both repos without duplicates
  -added CTI backend repo id to md file

### Why did you make the changes (we will use this info to test)?
  -the languages section now shows languages used in both CTI Frontend and CTI Backend 
 
### Screenshots of Proposed Changes Of The Website  (if any, please do not screen shot code changes)

<details>
<summary>Visuals before changes are applied</summary>

Screenshot from 2023-07-26 02-19-53

Screenshot from 2023-07-26 02-21-10

</details>

<details>

Screenshot from 2023-07-26 02-22-32

Screenshot from 2023-07-26 02-22-38

<summary>Visuals after changes are applied</summary> 
 

</details>

### CODE REVIEW

**Comment by StephenTheDev1001** [comment]
I don't like what I did on line 30 of project.js, but I can't think of a way to do it better without going beyond the scope of the issue.

File: _projects/civic-tech-index.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ---
 # 'identification' is the 9 digit ID for your repo in the GitHub API.
-identification: '241519642'
+identification: '241519642, 275296439'
```
Comment (MattPereira) [inline _projects/civic-tech-index.md]: adding the extra project id like this breaks the logic in `current-projects.js` which generates the project cards that show on the homepage and projects page

Screenshot from 2023-07-20 09-51-01

**Review by MattPereira — CHANGES_REQUESTED** [review]
@StephenTheDev1001 admire your persistence and hard work on this issue


you might consider adding an extra key/value pair to the `civic-tech-index.md` that contains the additional repo id and then just adding conditional logic to `projects.js` and `current-projects.js` that checks for if that property exists on the project object in order to display the extra languages on both the project details page and the project cards 

that way you may be able to avoid having to add nested loops to both files

**Comment by StephenTheDev1001** [comment]
@MattPereira Should I make it to where we can add more than one additional repo to scrape from or is one additional repo enough?

**Comment by MattPereira** [comment]
@StephenTheDev1001 I think one additional is fine since only this one project has multiple repos but I'm not up to date with website team affairs anymore 

@t-will-gillis @roslynwythe what do you think?

**Comment by t-will-gillis** [comment]
>@StephenTheDev1001 I think one additional is fine since only this one project has multiple repos but I'm not up to date with website team affairs anymore

@MattPereira I am not aware of any other project that would need more than one additional repo to scrape for language data.

File: _projects/civic-tech-index.md (hunk 1 above)
Comment (StephenTheDev1001) [reply _projects/civic-tech-index.md]: This helped a lot. Thanks

File: assets/js/project.js [hunk 2]
```diff
@@ -14,17 +14,28 @@ let projects = JSON.parse(decodeURIComponent("{{ projects | jsonify | uri_escape
...
+function findProjectById(identification){
+    // Starts at 1 now since the first element is a time stamp
+    for (let i = 1; i < projects.length; i++){
+        let itemId = projects[i].id.toString();
+        if(itemId == identification){
+            return projects[i]
+        }
     }
-}
+}
```
Comment (MattPereira) [inline assets/js/project.js:27]: I like the refactor here it's more readable with the function

Also using `const` and `return` feels much stronger than setting project to null and using `break` to end the loop :muscle:

File: assets/js/project.js [hunk 3]
```diff
@@ -14,17 +14,28 @@ let projects = JSON.parse(decodeURIComponent("{{ projects | jsonify | uri_escape
...
+} 
 
+// Merge the language sections if there's a second repo
+if (scriptTag.getAttribute("secRepoId")){
+    const secRepoId = scriptTag.getAttribute("secRepoId");    
+    const firstLangs = project.languages;
+    const secLangs = findProjectById(secRepoId).languages;
+
+    let languagesArr = [...firstLangs, ...secLangs];
+    let set = new Set(languagesArr);
```
Comment (MattPereira) [inline assets/js/project.js:36]: i think these could be `const` but its nbd. Nice use of `Set` !!! :tada:

**Review by MattPereira — APPROVED** [review]
Looks good @StephenTheDev1001 !!! 

Nice execution, solid communication with the team, and quick turnaround on the requested changes :+1:

**Comment by LRenDO** [comment]
Hi @adrianang! If you plan to review this PR, please add your ETA and availability when you have a minute. If not, please remove yourself as reviewer. Thanks!

**Comment by adrianang** [comment]
- Availability: FSS, 6 PM - 10 PM (PST) (Week of 7/24)
- ETA: I plan on finishing this issue by Saturday, 7/29 at 10 PM.

**Review by adrianang — APPROVED** [review]
Hi @StephenTheDev1001 — the branching is set up correctly, the corresponding issue is linked, and the requested change has been made (ensure that Civic Tech Index shows all languages between its two repositories). In testing your branch on my local environment through Docker, I see the same thing that you had screenshotted for both the card view on the main Website page and the CTI project page view itself, and no other project seems to be affected as a result of these code changes.

I like the guard clauses that you set up, as a result of adding the second repo ID instead of appending it onto `identification`, as it minimizes the number of nested loop calls that the function makes (if I recall correctly, one of your previous commits called nested loops on each project). Nice job optimizing!

Thank you for sticking with the issue and seeing it through to completion; apologies that I couldn't be of more help, but it's great that you were able to find a pretty clever solution. Your efforts are deeply appreciated! 🙌🏼

**Pull request merged**